lms/services/file.py

```diff
@@ -0,0 +1,23 @@
+from lms.models import File
+
+
+class FileService:
+    def __init__(self, application_instance, db):
+        self._application_instance = application_instance
+        self._db = db
+
+    def get(self, lms_id, type_):
+        return self._query.filter_by(lms_id=lms_id, type=type_).one_or_none()
+
+    @property
+    def _query(self):
+        return self._db.query(File).filter_by(
+            application_instance=self._application_instance
+        )
+
+
+def factory(_context, request):
+    return FileService(
+        application_instance=request.find_service(name="application_instance").get(),
```

Comment:
This is a pattern we use in most services now but I think is creates some problems to just avoiding passing one parameter around. 

I much rather do `File.service.get(application_instance, lms_id, type)`  especially if we added `application_instance` to request or context instead coupling each service instance to one AI.

If we were to use this service on for example the admin pages we would need to instantiate it manually once per AI which seems odd. Having a separate "AdminPagesFileService" seeps to defeat the purpose of the service in the first place if they are not reusable. 

In any case, none of this is relevant to this PR which just follows the established pattern so don't mind me.

Reply:
Ok, I've moved `application_instance` to a `get()` method argument. I think it might actually make the higher-level code slightly nicer too